### PR TITLE
Markdown linting: MD004

### PR DESCRIPTION
**Pull request by leslietilling**
## Purpose of this pull request

This pull request (PR) addresses #327  

To address MD004: 
https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md004---unordered-list-style

## Affected documentation pages

- https://docs.magento.com/m2/ee/user_guide/cms/editor-insert-link.html
- https://docs.magento.com/m2/ee/user_guide/cms/page-builder-add-content-block.html
- https://docs.magento.com/m2/ee/user_guide/payment/paypal-payflow-link.html
- https://docs.magento.com/m2/ee/user_guide/payment/paypal-payments-pro.html
- https://docs.magento.com/m2/ee/user_guide/payment/paypal-payments-standard.html
- https://docs.magento.com/m2/ee/user_guide/payment/purchase-order.html
- https://docs.magento.com/m2/ee/user_guide/payment/zero-subtotal-checkout.html
- https://docs.magento.com/m2/ee/user_guide/sales-channels/amazon/creating-editing-overrides.html
- https://docs.magento.com/m2/ee/user_guide/sales-channels/amazon/intelligent-repricing-rules.html
- https://docs.magento.com/m2/ee/user_guide/shipping/magento-shipping-batch-create.html

## Affected Magento editions

- [x] Open Source
- [x] Commerce
- [x] B2B

### CODE REVIEW

File: _checks/styles/style-rules-prod [hunk 1]
```diff
@@ -1,7 +1,7 @@
 rule 'MD001'
```
Comment (dshevtsov) [inline _checks/styles/style-rules-prod]:
```suggestion
rule 'MD001' # Header levels should only increment by one level at a time
```

File: _checks/styles/style-rules-prod [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 exclude_rule 'MD002'
 rule 'MD003', :style => :atx
```
Comment (dshevtsov) [inline _checks/styles/style-rules-prod]:
```suggestion
rule 'MD003', :style => :atx # Header style
```

File: _checks/styles/style-rules-prod [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
-exclude_rule 'MD004'
+rule 'MD004'
```
Comment (dshevtsov) [inline _checks/styles/style-rules-prod]:
```suggestion
rule 'MD004' # Unordered list style, default "consistent"
```

**Comment by leslietilling** [comment]
running tests

**Pull request merged**